domain/src/main/java/org/sagebionetworks/research/domain/form/ChoiceBase.java

```diff
@@ -165,43 +165,42 @@ protected boolean equalsHelper(Object o) {
      * @return A TypeToken<ChoiceBase<T>> with the same generic parameter as the given Type which should
      * be a Choice<T> for some Type T.
      */
-    private static TypeToken<ChoiceBase<?>> createChoiceBaseTypeToken(Type typeOfT) {
+    private static Type getInnerType(Type typeOfT) {
         TypeToken tToken = TypeToken.of(typeOfT);
         try {
             TypeToken token = tToken.resolveType(Choice.class.getDeclaredMethod("getAnswerValue")
                     .getGenericReturnType());
-            return createChoiceBaseTypeTokenHelper(token);
+            return token.getType();
         } catch (NoSuchMethodException e) {
             e.printStackTrace();
         }
 
         return null;
     }
 
-    /**
-     * Returns a TypeToken<ChoiceBase<genericToken.getType()>>
-     * @param genericToken The TypeToken for the class to be the generic type argument to ChoiceBase.
-     * @param <T> The Type of the generic type argument to ChoiceBase.
-     * @return a TypeToken<ChoiceBase<genericToken.getType()>>
-     */
-    private static <T> TypeToken<ChoiceBase<T>> createChoiceBaseTypeTokenHelper(TypeToken<T> genericToken) {
-        return new TypeToken<ChoiceBase<T>>() {}.where(new TypeParameter<T>() {}, genericToken);
-    }
-
-    public static JsonDeserializer<ChoiceBase<?>> getJsonDeserializer() {
-        return new JsonDeserializer<ChoiceBase<?>>() {
+    public static JsonDeserializer<ChoiceBase> getJsonDeserializer() {
+        return new JsonDeserializer<ChoiceBase>() {
             @Override
             public ChoiceBase deserialize(final JsonElement json, final Type typeOfT,
                     final JsonDeserializationContext context)
                     throws JsonParseException {
                 if (json.isJsonObject()) {
-                    // Suppose we are deserializing a Choice<T> for some type T. We create a TypeToken<ChoiceBase<T>>
-                    // then we deserialize using this.
-                    TypeToken token = createChoiceBaseTypeToken(typeOfT);
-                    ChoiceBase<?> result = context.deserialize(json, token.getType());
-                    if (result != null) {
-                        return result;
-                    }
+                    JsonObject object = json.getAsJsonObject();
```

Comment:
Does Gson's default behavior not work here?

Reply:
The default behavior would work except for the thing where { "value":"string", "text":"string" } and "string" should deserialize into the same ChoiceBase. If we removed support for that I believe I could just use Gson's default behavior. What are your thoughts on doing that?